Core/WebCacheManager.swift

```diff
@@ -82,14 +82,28 @@ public class WebCacheManager {
                       dataStoreIdManager: DataStoreIdManaging = DataStoreIdManager.shared) async {
 
         var cookiesToUpdate = [HTTPCookie]()
+        var leftoverContainerIDs = [UUID]()
         if #available(iOS 17, *) {
-            cookiesToUpdate += await containerBasedClearing(storeIdManager: dataStoreIdManager) ?? []
+            let result = await containerBasedClearing(storeIdManager: dataStoreIdManager)
+            cookiesToUpdate += result.cookies
+            leftoverContainerIDs = result.leftoverContainerIDs
         }
 
         // Perform legacy clearing to migrate to new container
         cookiesToUpdate += await legacyDataClearing() ?? []
 
         cookieStorage.updateCookies(cookiesToUpdate, keepingPreservedLogins: logins)
+
+        // Attempt to clean up leftover stores again after a delay
+        // This should not be a problem as these containers are not supposed to be used anymore.
+        // If this fails, we are going to still clean them next time as WebKit keeps track of all stores for us.
+        if #available(iOS 17, *), !leftoverContainerIDs.isEmpty {
+            DispatchQueue.main.asyncAfter(deadline: .now() + 3) {
```

Comment:
I'm a little worried about calling this on the main thread.  This function doesn't need to.  It will call back to the main thread, but the API suggests it can be called from any thread.  Also, I think it'd be better if this was doing as a task rather than DispatchQueue, e.g.

            Task {
                try? await Task.sleep(interval: 3.0)
                for uuid in leftoverContainerIDs {
                    try? await WKWebsiteDataStore.remove(forIdentifier: uuid)
                }
            }

That said, I'm going to approve because it does work so up to you if you want to tweak it.